### PR TITLE
closed #5118, remove WidgetType and refactor LayoutType to Type

### PR DESCRIPTION
**Pull request by zilongshanren**

### CODE REVIEW

File: cocos/editor-support/cocostudio/CCSGUIReader.cpp [hunk 1]
```diff
@@ -1234,9 +1249,22 @@ void WidgetPropertiesReader0300::setColorPropsForWidgetFromJsonDictionary(Widget
...
     widget->setColor(Color3B(colorR, colorG, colorB));
     bool apx = DICTOOL->checkObjectExist_json(options, "anchorPointX");
-    float apxf = apx ? DICTOOL->getFloatValue_json(options, "anchorPointX") : ((widget->getWidgetType() == Widget::Type::ELEMENT) ? 0.5f : 0.0f);
+    float apxf;
+    float apyf;
+    if (apx) {
+        apxf = DICTOOL->getFloatValue_json(options, "anchorPointX");
+    }else{
+        apxf = widget->getAnchorPoint().x;
+    }
```
Comment (minggo) [inline cocos/editor-support/cocostudio/CCSGUIReader.cpp]: It is better to use more meaningful names, such as:
apxf -> anchorPointX
apx -> isAnchorPointXInJson

It is not an urgent issue for it.

Comment (zilongshanren) [inline cocos/editor-support/cocostudio/CCSGUIReader.cpp]: I just keep the variable name as before. But i can fix it.

File: cocos/ui/UIWidget.cpp [hunk 2]
```diff
@@ -46,7 +46,6 @@ _touchEndPos(Vector2::ZERO),
 _touchEventListener(nullptr),
 _touchEventSelector(nullptr),
 _name("default"),
-_widgetType(Type::ELEMENT),
```
Comment (dumganhar) [inline cocos/ui/UIWidget.cpp:49]: Isn't `_widgetType` needed?

Comment (zilongshanren) [inline cocos/ui/UIWidget.cpp:49]: It is not needed. I confirm it with WenZhi cai

Comment (dumganhar) [inline cocos/ui/UIWidget.cpp:49]: OK

File: cocos/ui/UIWidget.h [hunk 3]
```diff
@@ -671,7 +656,6 @@ class Widget : public ProtectedNode
     ccWidgetTouchCallback _touchEventCallback;
     
     std::string _name;
-    Type _widgetType;
 	int _actionTag;
```
Comment (dumganhar) [inline cocos/ui/UIWidget.h]: Please fix the indention. Thanks.

Comment (zilongshanren) [inline cocos/ui/UIWidget.h]: ok, it's fixed now.

**Pull request merged**